src/core/fillOrder.ts

```diff
@@ -29,19 +35,100 @@ export const getTransactions = async (
     })
     .filter(Boolean) as Transaction[]
 
-  let fill
+  let fill: Transaction | undefined
 
-  const fillPayload = {
+  // Default fill payload from the bundle
+  const originalFillPayload = {
     to: bundle.targetFillPayload.to,
     data: bundle.targetFillPayload.data,
     chainId: bundle.targetFillPayload.chainId,
     value: 0n, // we never need to send value
   }
 
+  // Apply inventory-based rebalancing (if relayer address is provided)
+  if (relayerAddress && bundle.acrossDepositEvents.length > 0) {
+    // Process each deposit event individually
+    let optimalChainId: number | null = null
+
+    // Loop through each deposit event to find the optimal destination
+    for (const depositEvent of bundle.acrossDepositEvents) {
+      try {
+        // Skip if there's no origin claim payload (already filtered in claims)
+        if (depositEvent.originClaimPayload.chainId === 0) continue
+
+        // Determine optimal destination chain for this deposit event
+        const eventOptimalChainId = await getOptimalDestinationChain(
+          depositEvent,
+          relayerAddress,
+          getRPCUrl,
+        )
+
+        // If we found a valid optimal chain, consider it
+        if (eventOptimalChainId) {
+          // If we haven't found an optimal chain yet, use this one
+          if (!optimalChainId) {
+            optimalChainId = eventOptimalChainId
+            debugLog(
+              `Considering chain ${optimalChainId} as optimal destination`,
+            )
+          }
+          // If this deposit's optimal chain differs from a previous one,
+          // we have conflicting needs. In this simple implementation,
+          // we'll use the first valid one, but a more sophisticated approach
+          // might prioritize based on deficit size, value, etc.
+          else if (optimalChainId !== eventOptimalChainId) {
+            debugLog(
+              `Note: Deposit events have different optimal destinations (${optimalChainId} vs ${eventOptimalChainId})`,
+            )
+          }
+        }
+      } catch (error) {
+        // If processing one deposit fails, log and continue with others
+        debugLog(`Error processing deposit event: ${error}`)
+        // We continue the loop to process other deposit events
+      }
+    }
+
+    // If we found an optimal chain different from the original destination
+    if (optimalChainId && optimalChainId !== bundle.targetFillPayload.chainId) {
+      debugLog(
+        `Rebalancing: Redirecting fill from chain ${bundle.targetFillPayload.chainId} to chain ${optimalChainId}`,
+      )
+
+      // TODO: In a real implementation, we would modify the fill transaction here
+      // to target the new chain. This would involve:
+      // 1. Generate a new transaction for the optimal chain
```

Comment:
what do you mean here?

Reply:
i think this was leftover from a previous commit, will remove